File: src/main/java/com/mpe85/grampa/event/MatchFailureEvent.java
package com.mpe85.grampa.event;

import com.mpe85.grampa.rule.RuleContext;

/**
 * Event posted on a match failure.
 * 
 * @author mpe85
 *
 * @param <T>
 *            the type of the stack elements
 */
public class MatchFailureEvent<T> extends RuleContextEvent<T> {
	
	public MatchFailureEvent(final RuleContext<T> context) {
		super(context);
	}
	
}

File: src/main/kotlin/com/mpe85/grampa/event/MatchFailureEvent.kt
package com.mpe85.grampa.event

import com.mpe85.grampa.rule.RuleContext

/**
 * Event posted on a match failure.
 *
 * @author mpe85
 * @param[T] The type of the stack elements
 * @param[context] The rule context in which the event was posted
 */
class MatchFailureEvent<T>(context: RuleContext<T>) : RuleContextEvent<T>(context)
